ONSdigital/dp-frontend-filter-dataset-controller
# Fix/latest version editions specific

### What

Currently, when users are on the filter journey for an older version of a CMD dataset, the links that take them to the latest version currently do so irrespective of edition. E.g., a user could be on the 2010-19 edition of weekly deaths v1, and when they click to apply the filters to the latest release, it creates a new filter job and redirects them to the latest version of the *covid-19* edition of the dataset. This is unintended, and can also cause an error, because the dimensions available can be different between editions.

This PR fixes the implementation so that it only renders/redirects if there is a more recent version of the same edition the user is in.

This PR also implements a new test case for the `UseLatest()` method as there wasn't one previously, as well as updates ONS dependencies

### How to review

- Sense check code and implementation
- Ensure tests pass

#### To test locally

- You need a dataset with multiple editions and versions locally. You can use the [weekly deaths data](https://www.ons.gov.uk/datasets/weekly-deaths-age-sex) csvs, if not already set up
- Go through the dataset publishing journey in Florence to create at least 2 versions of both editions
- Visit the filter overview page of v1 for each edition
- You should see the warning component. "View latest release" should link you to the latest version of that edition
- Apply some filters and click on "Apply these filters to the latest release". You should now be on a new filter job, with the same dimensions listed, and the filters you'd applied previously still persisted

**v1 covid-19 edition**
![image](https://user-images.githubusercontent.com/23668262/98515542-1a1ac200-2263-11eb-8e5a-d1410ed76d30.png)

**v2 covid-19 edition after clicking "Apply these filters to the latest release"**
![image](https://user-images.githubusercontent.com/23668262/98515596-29017480-2263-11eb-941b-4b1d0bb2c517.png)

### Who can review

Anyone
